crates/intrinsic-test/src/arm/mod.rs

```diff
@@ -69,28 +69,35 @@ impl SupportedArchitectureTest for ArmArchitectureTest {
 
         let (chunk_size, chunk_count) = chunk_info(self.intrinsics.len());
 
-        let cpp_compiler = compile::build_cpp_compilation(&self.cli_options).unwrap();
+        let cpp_compiler = compile::build_cpp_compilation(&self.cli_options);
+
+        match fs::exists("c_programs") {
+            Ok(false) => fs::create_dir("c_programs").unwrap(),
+            Ok(true) => {}
+            _ => return false,
+        }
```

Comment:
I believe `create_dir_all` has the semantics that you want here (just return succes if the path already exists)